### DISCUSSION TITLE
Mac OS 12+ virtualize on M2 iPad

### DISCUSSION THREAD

**Discussion by jinh9015** [utmapp/UTM#6253] [macOS]
I have two Apple Silicon devices, M2 MacBook Air (Ventura) and M2 iPad Pro (iOS16.2).

First, my iPad is still on iOS16.2(which is definitely below 16.3.1) so hypervisor works.

I'm trying to virtualize Mac OS 12+ on my iPad.

It worked fine with Macbook UTM. It had Mac OS 12~ virtualize option with restore ipsw file and successfully created working VM.

However, on M2 iPad, there is only Windows 10+ and Linux options for virtualize, so cannot try with same setting.

Also, it is not posiible to copy complete .utm from Macbook to iPad. UTM on iPad doesn't recognize it as loadable package.

Is it because Mac 12+ virtualization is not using qemu but OS local kernel hypervisor?

Is it not possible at all to virtualize Mac OS 12+ on M2 iPad?

**Comment by nfzerox** [comment]
Check out: https://github.com/nfzerox/VirtualMacOniPad
